airbyte-integrations/connectors/source-retently/source_retently/source.py

```diff
@@ -11,15 +11,30 @@
 from airbyte_cdk.sources import AbstractSource
 from airbyte_cdk.sources.streams import Stream
 from airbyte_cdk.sources.streams.http import HttpStream
-from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.sources.streams.http.auth import Oauth2Authenticator, TokenAuthenticator
 
 
 class SourceRetently(AbstractSource):
+    @staticmethod
+    def get_authenticator(config):
+        credentials = config.get("credentials", {})
+        if credentials and "client_id" in credentials:
+            return Oauth2Authenticator(
+                token_refresh_endpoint="https://app.retently.com/api/oauth/token",
+                client_id=credentials.get("client_id"),
+                client_secret=credentials.get("client_secret"),
+                refresh_token=credentials.get("refresh_token"),
```

Comment:
these options are required. you can use their direct call without `get`

Reply:
nice catch, fixed